docs/git.md

```diff
@@ -0,0 +1,74 @@
+# Git Workflow
+
+## Table Of Contents
+
+[Git Commit Guidelines](#commit)
+  - [Commit Message Format](#format)
+  - [Type](#type)
+  - [Subject](#subject)
+  - [Emoji](#emoji)
+  - [Body](#body)
+
+
+## <a name="commit"></a> Git Commit Guidelines
+
+We have very precise rules over how our git commit messages can be formatted. This leads to **more
+readable messages** that are easy to follow when looking through the **project history**.  But also,
+we use the git commit messages to **create change log**.
+
+
+### <a name="format"></a> Commit Message Format
+
+Each commit message consists of a **header** and a **body**. The header has a special
+format that includes a **type**, a **subject** and an **emoji**:
+
+```
+<type>: <subject> (<emoji>)
+<BLANK LINE>
+<body>
+```
+
+The header is mandatory and the emoji of the header is optional. (Limit to 50 characters or less<sup>[ref](https://chris.beams.io/posts/git-commit/#limit-50)</sup>)
+
+Any line of the commit message cannot be longer 72 characters
+<sup>[ref](https://chris.beams.io/posts/git-commit/#wrap-72)</sup>!
+This allows the message to be easier to read on GitHub as well as in various git tools.
+
+
+### <a name="type"></a> Type
+
+Must be one of the following
+<sup>[ref](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#type)</sup>:
```

Comment:
Was this document taken from the angular repository? If so, it should be credited!